### PR TITLE
Add models/views/tests for routes

### PR DESCRIPTION
**Pull request by asaunier**
It's mostly copy-pasted from the waypoints classes, adapting the attributes.
Next step is to factorize the common parts between waypoints, routes, etc. I think.

### CODE REVIEW

File: app_api/models/route.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    'snow_ice_mixed',
+    'mountain_climbing',
+    'rock_climbing',
+    'ice_climbing',
+    'hiking',
+    'snowshoeing',
+    'paragliding',
+    'mountain_biking',
+    'via_ferrata'
+    ]
```
Comment (elemoine) [inline app_api/models/route.py]: Is this actually used?

File: app_api/models/route.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        copy_attributes(self, route, Route._ATTRIBUTES)
+
+        return route
+
+    def get_archive_locales(self):
+        locales = []
+
+        for locale in self.locales:
+            archive_local = locale.to_archive()
+            locales.append(archive_local)
```
Comment (elemoine) [inline app_api/models/route.py]: How about using a list comprehension?

```py
    def get_archive_locales(self):
        return [locale.to_archive() for locale in self.locales]
```

File: app_api/models/route.py (hunk 1 above)
Comment (asaunier) [inline app_api/models/route.py]: Yes i'ts used at line 34.
"Activities" is one of the main attributes of routes and outings. See for instance: http://www.camptocamp.org/routes/54417/fr/petite-dent-de-veisivi-traversee-des-aretes

By the way routes/outings may have several activities at the same time (for instance: http://www.camptocamp.org/routes/46065/fr/cheval-blanc-depuis-emosson)
=> not sure the current Enum type is correct?

Comment (elemoine) [inline app_api/models/route.py]: Not sure either. I guess this can be changed later.

Comment (asaunier) [inline app_api/models/route.py]: Yep, that's what I thought when adding the attribute. :)

File: app_api/tests/__init__.py [hunk 3]
```diff
@@ -61,6 +61,10 @@ def setUp(self):  # noqa
         self.trans = connection.begin()
 
         # bind an individual Session to the connection
+        # Next line is needed to make several tests run in a row.
+        # See https://github.com/Pylons/webtest/issues/5
+        # FIXME Is there a better solution?
+        DBSession.remove()
```
Comment (elemoine) [inline app_api/tests/__init__.py]: That part is weird. But that entire setup process looks weird to me. I'd like to take a look at it, but I suggest that you merge your branch first.

**Comment by asaunier** [comment]
OK thanks @elemoine !
Merging

**Pull request merged**